docs/guides/debugging-flag-assignment.md

```diff
@@ -0,0 +1,222 @@
+      "allocationEvaluationCode": "MATCH",
+      "orderPosition": 2
+    },
+    "unmatchedAllocations": [
+      {
+        "key": "allocation-6647",
+        "allocationEvaluationCode": "FAILING_RULE",
+        "orderPosition": 1
+      }
+    ],
+    "unevaluatedAllocations": [
+      {
+        "key": "allocation-6648",
+        "allocationEvaluationCode": "UNEVALUATED",
+        "orderPosition": 3
+      }
+    ]
+  }
+}
+```
+
+### Scenario: An allocation was matched due to a matching rule
+
+Let's see what happens when we specify `{ companyId: 11 }` as the `subjectAttributes`. This should match the **Alpha Testers** allocation, which ensures `test` is assigned for `companyId` values of `11` or `13`.
+
+In this case, the `matchedAllocation` field contains `"orderPosition": 1`, which means that the **Alpha Testers** allocation was matched, since that's the first allocation defined in our configuration. The `matchedRule` field gives us information about which rule was matched, and `flagEvaluationDescription` tells us that the match happened due to matching rules. In this scenario, all remaining allocations that are defined in our configration will be listed in `unevaluatedAllocations`, since the first allocation was matched.
+
+```typescript
+const flagKey = 'some-disabled-flag';
+const subjectKey = 'subject-123'
+const subectAttributes = { companyId: 11 }; // companyId of 11 includes alpha testers
+const defaultValue = 'default'
+
+getStringAssignmentDetails(flagKey, subjectKey, subjectAttributes, defaultValue)
+// returns =>
+{
+  "value": "test",
+  "evaluationDetails": {
+    "environmentName": "Production",
+    "flagEvaluationCode": "MATCH",
+    "flagEvaluationDescription": "Supplied attributes match rules defined in allocation \"allocation-6647\".",
+    "variationKey": "test",
+    "variationValue": "test",
+    "configFetchedAt": "2024-07-09T14:25:44.441Z",
+    "configPublishedAt": "2024-07-09T13:44:02.584Z",
+    "matchedRule": {
+      "conditions": [
+        {
+          "attribute": "companyId",
+          "operator": "ONE_OF",
+          "value": ["11", "13"]
+        }
+      ]
+    },
+    "matchedAllocation": {
+      "key": "allocation-6647",
+      "allocationEvaluationCode": "MATCH",
+      "orderPosition": 1
+    },
+    "unmatchedAllocations": [],
+    "unevaluatedAllocations": [
+      {
+        "key": "allocation-5055",
+        "allocationEvaluationCode": "UNEVALUATED",
+        "orderPosition": 2
+      },
+      {
+        "key": "allocation-6648",
+        "allocationEvaluationCode": "UNEVALUATED",
+        "orderPosition": 3
+      }
+    ]
+  }
+}
+
+```
+
+### Scenario: Your flag is disabled
+
+When your flag is disabled, your `flagEvaluationCode` will be `FLAG_UNRECOGNIZED_OR_DISABLED`, and your supplied `defaultValue` will be assigned to `value`. Pay attention to your `environmentName` in this scenario since you may not be working in the environment you used to define your allocations.
```

Comment:
Consider changing "your allocations" to "your flags" at the end of the last sentence.

Reply:
updated!